File: anhilist.py
import xml.etree.ElementTree as ET
import re
import argparse

def create_anilist_compatible_xml(input_file, output_file):
    tree = ET.parse(input_file)
    root = tree.getroot()
    
    myanimelist = ET.Element("myanimelist")
    
    for folder in root.findall('.//folder'):
        folder_name = folder.find('name').text
        
        for item in folder.find('data').findall('item'):
            name = item.find('name').text
            link = item.find('link').text
            
            mal_id = re.search(r'/anime/(\d+)', link)
            mal_id = mal_id.group(1) if mal_id else "0"
            
            anime = ET.SubElement(myanimelist, "anime")
            
            ET.SubElement(anime, "series_animedb_id").text = mal_id
            ET.SubElement(anime, "series_title").text = name
            
            status_map = {
                "Watching": "1",
                "Completed": "2",
                "On-Hold": "3",
                "Dropped": "4",
                "Plan to watch": "6"
            }
            ET.SubElement(anime, "my_status").text = status_map.get(folder_name, "6")
            
            ET.SubElement(anime, "my_score").text = "0"
            ET.SubElement(anime, "my_watched_episodes").text = "0"
            ET.SubElement(anime, "my_start_date").text = "0000-00-00"
            ET.SubElement(anime, "my_finish_date").text = "0000-00-00"
            ET.SubElement(anime, "my_rewatching").text = "0"
            ET.SubElement(anime, "my_rewatching_ep").text = "0"
            ET.SubElement(anime, "update_on_import").text = "1"
    
    tree = ET.ElementTree(myanimelist)
    tree.write(output_file, encoding="UTF-8", xml_declaration=True)
    
    return True

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description='Convert custom anime list to MAL format for AniList import')
    parser.add_argument('input_file', help='Input XML file')
    parser.add_argument('-o', '--output', help='Output filename', default='myanimelist_export.xml')
    
    args = parser.parse_args()
    
    success = create_anilist_compatible_xml(args.input_file, args.output)
    
    if success:
        print(f"Conversion completed. Output saved to {args.output}")

File: anhilist.rb
require 'nokogiri'
require 'optparse'

def create_anilist_compatible_xml(input_file, output_file)
  doc = File.open(input_file) { |f| Nokogiri::XML(f) }
  
  builder = Nokogiri::XML::Builder.new(encoding: 'UTF-8') do |xml|
    xml.myanimelist {
      doc.xpath('//folder').each do |folder|
        folder_name = folder.at_xpath('./name').text
        
        folder.xpath('./data/item').each do |item|
          name = item.at_xpath('./name').text
          link = item.at_xpath('./link').text
          
          mal_id = link.match(/\/anime\/(\d+)/)
          mal_id = mal_id ? mal_id[1] : "0"
          
          xml.anime {
            xml.series_animedb_id mal_id
            xml.series_title name
            
            status_map = {
              "Watching" => "1",
              "Completed" => "2",
              "On-Hold" => "3",
              "Dropped" => "4",
              "Plan to watch" => "6"
            }
            xml.my_status status_map[folder_name] || "6"
            
            xml.my_score "0"
            xml.my_watched_episodes "0"
            xml.my_start_date "0000-00-00"
            xml.my_finish_date "0000-00-00"
            xml.my_rewatching "0"
            xml.my_rewatching_ep "0"
            xml.update_on_import "1"
          }
        end
      end
    }
  end
  
  File.write(output_file, builder.to_xml)
  true
end

options = {}
OptionParser.new do |opts|
  opts.banner = "Usage: convert_to_mal.rb INPUT_FILE [options]"
  
  opts.on("-o", "--output FILENAME", "Output filename") do |o|
    options[:output] = o
  end
end.parse!

input_file = ARGV[0]
output_file = options[:output] || "myanimelist_export.xml"

if input_file.nil?
  puts "Error: Input file is required"
  puts "Usage: convert_to_mal.rb INPUT_FILE [-o OUTPUT_FILE]"
  exit 1
end

success = create_anilist_compatible_xml(input_file, output_file)

if success
  puts "Conversion completed. Output saved to #{output_file}"
end
